radixdlt/src/main/java/org/radix/utils/Duration.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package org.radix.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * Utility class convenient for representation of fixed durations in cases when both,
+ * value and time unis are necessary, for example, with
+ * {@link java.util.concurrent.CountDownLatch#await(long, TimeUnit)}).
+ * <p>
+ * Provided {@link #parse(String)} method is able to parse strings which represent a duration.
+ * Recognized string format:
+ * <pre>
+ *     [0-9]+(s|m|h)?
+ * </pre>
+ * Leading number represents value duration and must be greater than zero. Fractional values are not supported.
+ * Optional trailing character defines units of duration value:
+ * <ul>
+ *     <li><b>s</b> - second(s)</li>
+ *     <li><b>m</b> - minute(s)</li>
+ *     <li><b>h</b> - hour(s)</li>
+ * </ul>
+ * If units are omitted or not recognized then <b>seconds</b> are assumed by default.
+ */
+public class Duration {

```

Comment:
I'm not sure most of this class is required?  Couldn't we just implement a DurationParser that returns a java.time.Duration?  That's less code that we need to maintain.

Reply:
Done